### PR TITLE
ng15 with eslint

### PR DESCRIPTION
**Pull request by jornare**
Updated to Angular 15
Replaced deprecated tslint with new eslint
Removed unused protractor
Updated test script (type error)

### CODE REVIEW

File: packages/icons-angular/package.json [hunk 1]
```diff
@@ -4,41 +4,48 @@
   "scripts": {
     "ng": "ng",
     "start": "ng serve",
-    "lint": "ng lint",
+    "lint": "ng lint --project ant-design-icons-angular",
     "test": "ng test --no-watch --no-progress --browsers=ChromeHeadlessCI --code-coverage",
-    "build": "ng build --prod",
-    "build:lib": "npm run generate && ng build ant-design-icons-angular --configuration production && npm run copy:assets",
+    "build": "ng build --project ant-design-icons-angular --configuration production",
```
Comment (wzhudev) [inline packages/icons-angular/package.json]: This script is for building the demo webpage. Please do not change this.

**Comment by wzhudev** [comment]
Thanks for your PR!

File: packages/icons-angular/package.json (hunk 1 above)
Comment (jornare) [reply packages/icons-angular/package.json]: ng build --prod is deprecated.

Comment (jornare) [reply packages/icons-angular/package.json]: As is default project. It has to be named.

Comment (jornare) [reply packages/icons-angular/package.json]: If I chose the wrong project, I'll fix it

Comment (jornare) [reply packages/icons-angular/package.json]: fixed the scripts. also removed dependency to core-js, which is deprecated from Angular a long time ago and also not in use.

**Comment by jornare** [comment]
I see the react native tests are failing, but I did not and I'm not going to touch those parts.

**Comment by jornare** [comment]
I see that my PR is requesting to be merged directly to master. I'm not sure if that's appropriate. Please verify and change if needed.

**Review by wzhudev — APPROVED** [review]
LGTM

**Comment by wzhudev** [comment]
> I see the react native tests are failing, but I did not and I'm not going to touch those parts.

Please leave that to us. :p

Thanks for your pull request again. Much appreciated. The version and target branch are ok. We will make a release soon.

File: packages/icons-angular/package.json (hunk 1 above)
Comment (wzhudev) [reply packages/icons-angular/package.json]: Looks good!

**Pull request merged**